### PR TITLE
North-West Class-4-Abed Al-JavaScript-Core-2-Coursework-Week1

### PR DESCRIPTION
**Pull request by Abed-Aghbar**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: mandatory/1-writers.js [hunk 1]
```diff
@@ -66,9 +66,11 @@ Exercise 1:
 
   "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
 */
-function logAllWriters() {
+function logAllWriters(arr) {
   // write your code to log all writers here
-};
+  
+  arr.forEach(ele => console.log("Hi, my name is" + ele.firstName + " " + ele.lastName
```
Comment (mike-shields-cyf) [inline mandatory/1-writers.js:72]: Good work, but please be mindful of your formatting.

File: mandatory/1-writers.js [hunk 2]
```diff
@@ -79,10 +81,41 @@ Exercise 2:
   "Writer {firstName} {lastName} died at {age} years old."
 */
 
-function logDeadWritersInTheirForties() {
+function logDeadWritersInTheirForties(writers) {
   // write your code here
+
+writers.filter(keys => keys.age >= 40 && keys.age <= 49 && keys.alive === false)
```
Comment (mike-shields-cyf) [inline mandatory/1-writers.js:87]: Good work Abed, again be mindful of your formatting. As a general rule no more than 80 characters per line.

File: mandatory/1-writers.js [hunk 3]
```diff
@@ -91,10 +124,15 @@ Exercise 3:
...
+function logAliveWritersInTheirForties(writers) {
   // write your code here
+  return writers
+  .filter((keys) => keys.age >= 40 && keys.age <= 49 && keys.alive === true)
+  .map(
+    (keys) => `Hi, my name is ${keys.firstName} ${keys.lastName}. I am ${keys.age} years old.`
+  );
 }
-
+console.log(logAliveWritersInTheirForties(writers));
```
Comment (mike-shields-cyf) [inline mandatory/1-writers.js:135]: The logic above is correct but like `logDeadWritersInTheirForties` a `console.log()` should be executed for each object in the array.

File: mandatory/10-cheap-diner.js [hunk 4]
```diff
@@ -30,6 +30,14 @@ Should give the answer "Nothing :("
 **/
 
 function chooseMeal(mealArray) {
```
Comment (mike-shields-cyf) [inline mandatory/10-cheap-diner.js:32]: Hi Abed, this isn't correct, make sure you review the solutions repo.

File: mandatory/2-eligible-students.js [hunk 5]
```diff
@@ -20,6 +20,7 @@
  */
 
 function eligibleStudents(attendances) {
```
Comment (mike-shields-cyf) [inline mandatory/2-eligible-students.js:22]: Good work Abed. Again be mindful of formatting, this is a very long line of code.

File: mandatory/3-journey-planner.js [hunk 6]
```diff
@@ -27,8 +27,14 @@
 */
 
 function journeyPlanner(locations, transportMode) {
```
Comment (mike-shields-cyf) [inline mandatory/3-journey-planner.js:29]: Good work Abed

File: mandatory/4-water-bottle.js [hunk 7]
```diff
@@ -24,18 +24,23 @@ let bottle = {
   volume: 0,
   fillUp: function () {
     // calling this function should completely fill your bottle (volume = 100);
+  this.volume = 100;
   },
   pour: function () {
```
Comment (mike-shields-cyf) [inline mandatory/4-water-bottle.js:29]: `pour` allows us to exceed a `volume` of `100`. How could this be prevented?

File: mandatory/4-water-bottle.js [hunk 8]
```diff
@@ -24,18 +24,23 @@ let bottle = {
...
     // calling this function should increase your bottle volume by 10 units;
+  this.volume += 10;
   },
   drink: function () {
```
Comment (mike-shields-cyf) [inline mandatory/4-water-bottle.js:33]: `drink` allows us to have a negative `volume`. How could this be prevented?

File: mandatory/5-groceries.js [hunk 9]
```diff
@@ -29,14 +29,29 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
-
+Object.keys(weeklyMealPlan).forEach((key) => {
```
Comment (mike-shields-cyf) [inline mandatory/5-groceries.js:32]: Nice work Abed.

File: mandatory/5-groceries.js [hunk 10]
```diff
@@ -29,14 +29,29 @@ Exercise 1:
...
+  });
+});
 /*
 Exercise 2:
   Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
 */
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
-
+for (day in weeklyMealPlan) {
```
Comment (mike-shields-cyf) [inline mandatory/5-groceries.js:45]: Nice work Abed.

File: mandatory/5-groceries.js [hunk 11]
```diff
@@ -53,7 +68,9 @@ let numberOfItemsPerWeek = {
   saturday: 0,
   sunday: 0,
 };
-
+for (i in numberOfItemsPerWeek) {
```
Comment (mike-shields-cyf) [inline mandatory/5-groceries.js:71]: Good work, try to use better variable names though, `day` and `groceryList` would be better than `i` and `i`

File: mandatory/6-people-I-know.js [hunk 12]
```diff
@@ -383,7 +383,8 @@ First, I want you to find all of my friends who are 35 or older.
 */
 
 let thirtyFiveOrOlder = [];
```
Comment (mike-shields-cyf) [inline mandatory/6-people-I-know.js:385]: Hi Abed, your logic is ok. Just some syntax errors. `let` variables cannot be redeclared. All you need to do here is assign to `thirtyFiveOrOlder` like this: 

```
let thirtyFiveOrOlder = [];
thirtyFiveOrOlder = friends.filter((person) => person.age >= 35);
```

Or you can just do it all in one go like this: 
```
let thirtyFiveOrOlder = friends.filter((person) => person.age >= 35);
```

`return` is only valid inside a function. 

If you correct your code by following the above advice. It passes all the tests.

File: mandatory/6-people-I-know.js [hunk 13]
```diff
@@ -383,7 +383,8 @@ First, I want you to find all of my friends who are 35 or older.
...
-
+let thirtyFiveOrOlder = friends.filter((person) => person.age >= 35);
+return thirtyFiveOrOlder;
```
Comment (mrmikeshields-cyf) [inline mandatory/6-people-I-know.js:387]: `return` will only work inside a function. There is no need to `return` here anyway. The tests will check that `thirtyFiveOrOlder` is the correct value.

File: mandatory/7-recipes.js [hunk 14]
```diff
@@ -24,5 +24,33 @@ You should write and log at least 5 recipes
...
+    title: "Food_4",
+    Serves: 2,
+    Ingredients: ['peas','sweet potato','corn']
+};
+
+let recipes_5 = {
+    title: "Food_5",
+    Serves: 2,
+    Ingredients: ['bean','mushrooms','corn']
+};
```
Comment (mike-shields-cyf) [inline mandatory/7-recipes.js:56]: Hi Abed, you've made a good start. Please check the solutions repo to see how to solve this exercise.

File: mandatory/8-reading-list.js [hunk 15]
```diff
@@ -24,7 +24,23 @@ without using any variables or any logic like loops, template strings or if stat
...
+  {title: 'The Cat in The Hat',
+  author:'Dr Seuss',
+  alreadyRead: false},
+  {title: 'Charlotte Web',
+  author:'E.B White',
+  alreadyRead: true}
+];
   
 // exercise 1
 function logBooks() {
```
Comment (mrmikeshields-cyf) [inline mandatory/8-reading-list.js:46]: Abed, I'm sure you can manage this task. Check the solutions repo to see how to solve this one.

File: mandatory/8-reading-list.js [hunk 16]
```diff
@@ -24,7 +24,23 @@ without using any variables or any logic like loops, template strings or if stat
 
 */
 
-const books = [];
+const books = [
```
Comment (mike-shields-cyf) [inline mandatory/8-reading-list.js:27]: This challenge is poorly written and confusing. You have made a good attempt at it as I can see that your logic is correct.

**Pull request closed without merging**